docs: contributing docs updates (#3734)
carbon-design-system/carbon-website: src/pages/contributing/documentation.mdx

-->

## Development considerations

Keep these considerations in mind if you are modifying Carbon or creating a
custom component.

<!--
Provide a bulleted list of implementation considerations for ensuring accessibility. If there is the ability to refernce the IBM toolkit or the ARIA authoring practices, that should be done.

- Use dashed bullets and ensure code is styled as code and consistently appears with the
following symbols: <element>, 'variable' and "value".
- End each bulleted list item with a period.

Example:
- See the [ARIA authoring practices](https://w3c.github.io/aria-practices/#tooltip) for more considerations.

-->
```

## Documenting patterns

Patterns show reusable combinations of components and templates that address
common user objectives with sequences and flows. Because they are more
complicated than components, you may need to adjust the topic order to best tell
the story — but your pattern should cover all of the topics outlined in the
following templates.

### Pattern template: for one variant

Use this template for documenting patterns that have a single variant. You can

> see this template in use for [Notification](/patterns/notification-pattern/).

Author: ```suggestion
see this template in use for [Toolbar](/patterns/text-toolbar-pattern/).
```

---
title: Documentation
description:
  All components and patterns require usage, style, code, and accessibility
  guidance on the Carbon website. Check out the markdown templates and
  instructions below to contribute documentation smoothly.
---

export const Title = () => <span>Contribute docs</span>;

> ;

Reviewer: ```suggestion

```

### Headings
Carbon accordions are not set as headings by default. For improved accessibility, annotate accordions as headings on the first occurrence in a product. Annotate the heading level of accordions as needed.

-->

## Development considerations

Keep these considerations in mind if you are modifying Carbon or creating a
custom component.

<!--
Provide a bulleted list of implementation considerations for ensuring accessibility. If there is the ability to refernce the IBM toolkit or the ARIA authoring practices, that should be done.

- Use dashed bullets and ensure code is styled as code and consistently appears with the
following symbols: <element>, 'variable' and "value".
- End each bulleted list item with a period.

Example:
- See the [ARIA authoring practices](https://w3c.github.io/aria-practices/#tooltip) for more considerations.

-->
```

## Documenting patterns

Patterns show reusable combinations of components and templates that address
common user objectives with sequences and flows. Because they are more
complicated than components, you may need to adjust the topic order to best tell
the story — but your pattern should cover all of the topics outlined in the

> following templats.

Reviewer: ```suggestion
following templates.
```

### Pattern template: for one variant

Use this template for documenting patterns that have a single variant. You can

> see this template in use for [Notification](/patterns/notification-pattern/).

Reviewer: I would maybe use Text toolbar here as a "one variant" example. This is tricky, because as you said, we really are not consistent with how we treat our anchor links for these patterns. The Notification pattern does have variants or types though which is why I am suggesting a more simple pattern like Text toolbar.

- Use sentence case for everything, including component names, e.g. Content
  switcher and Data table.

## Publishing updates

Ready to get your work reviewed and published? There are two ways to do this,

> depending on your experience with Github. You may also draft your content and

Reviewer: ```suggestion
depending on your experience with GitHub. You may also draft your content and
```

bring it to
[office hours](https://ec.yourlearning.ibm.com/w3/series/10289694?layout=grid&_ga=2.129990351.257088430.1694013487-309583455.1691093939&_gl=1*16l2j5q*_ga*MzA5NTgzNDU1LjE2OTEwOTM5Mzk.*_ga_FYECCCS21D*MTY5NDAzMjgwMi4xMS4xLjE2OTQwMzQ0OTEuMC4wLjA.)
to get further guidance.

### Editing in browser

#### How to edit an existing page

If you are editing an existing docs page, the easiest option is to click on the

> `Edit this page on Github` link that appears at the bottom of every page on the

Reviewer: ```suggestion
`Edit this page on GitHub` link that appears at the bottom of every page on the
```

> site. The link opens the specific Github page where you can edit the content and

Reviewer: ```suggestion
site. The link opens the specific GitHub page where you can edit the content and
```
